SpatialGDK/Source/SpatialGDKTests/SpatialGDK/EngineClasses/SpatialVirtualWorkerTranslator/SpatialVirtualWorkerTranslatorTest.cpp

```diff
@@ -1,41 +1,81 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
+#include "LBStrategySpy.h"
+
 #include "Tests/TestDefinitions.h"
 
 #include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "SpatialConstants.h"
 #include "Utils/SchemaUtils.h"
-#include "UObject/UObjectGlobals.h"
 
-#include "CoreMinimal.h"
+#include "Templates/UniquePtr.h"
+#include "UObject/UObjectGlobals.h"
 
+#include <WorkerSDK/improbable/c_worker.h>
 #include <WorkerSDK/improbable/c_schema.h>
 
 #define VIRTUALWORKERTRANSLATOR_TEST(TestName) \
 	GDK_TEST(Core, SpatialVirtualWorkerTranslator, TestName)
 
-VIRTUALWORKERTRANSLATOR_TEST(Given_init_is_not_called_THEN_return_not_ready)
+VIRTUALWORKERTRANSLATOR_TEST(GIVEN_init_is_not_called_THEN_return_not_ready)
+{
+	ULBStrategySpy* LBStrategySpy = NewObject<ULBStrategySpy>();
+	TUniquePtr<SpatialVirtualWorkerTranslator> Translator = MakeUnique<SpatialVirtualWorkerTranslator>(LBStrategySpy, SpatialConstants::TRANSLATOR_UNSET_PHYSICAL_NAME);
+
+	TestFalse("Translator without local virtual worker ID is not ready.", Translator->IsReady());
+	TestTrue("LBStrategy spy reports an invalid virtual worker ID.", LBStrategySpy->GetVirtualWorkerId() == SpatialConstants::INVALID_VIRTUAL_WORKER_ID);
+
+	return true;
+}
+
+VIRTUALWORKERTRANSLATOR_TEST(GIVEN_worker_name_specified_in_constructor_THEN_return_correct_local_worker_name)
+{
+	TUniquePtr<SpatialVirtualWorkerTranslator> Translator = MakeUnique<SpatialVirtualWorkerTranslator>(nullptr, "my_worker_name");
+
+	TestTrue("Local physical worker name returned correctly.", Translator->GetLocalPhysicalWorkerName() == "my_worker_name");
```

Comment:
```suggestion
	TestEqual<FString>("Local physical worker name returned correctly", Translator->GetLocalPhysicalWorkerName(), "my_worker_name");
```
Maybe? You may have to wrap some things in FString, but should provide better error information if it fails... although I don't expect it to fail often.